### PR TITLE
Fix automated releases

### PR DESCRIPTION
**Pull request by poncovka**
* Never run scheduled workflows on forks.
* Set up the access token during the code checkout.
* Fix the generated changelog. It is necessary to fetch all tags.
* Fix the condition for bumping a release version.

Tested with a fine-grained personal access token (Repository permissions -> Contents  -> Access: Read and write) on a protected branch in my fork: [successful run](https://github.com/poncovka/anaconda/actions/runs/3541904444/jobs/5946749334), [aborted run](https://github.com/poncovka/anaconda/actions/runs/3541913472/jobs/5946769916), [release commit](https://github.com/poncovka/anaconda/commit/52213d2a724828e0585f7ff8bbaec1f794972d15)

### CODE REVIEW

File: .github/workflows/release-periodically.yml [hunk 1]
```diff
@@ -45,21 +45,20 @@ jobs:
           # 128 -> fatal: no tag exactly matches 'c88a1272ffbc002042fd62ed409734c1c150cc5f'
           # 0 -> anaconda-38.6-1
           if git describe --tags --exact-match ; then
-            echo "Already tagged at HEAD, will not push another tag."
-            echo "::set-output name=can-tag::false"
+            echo "::warning:: Already tagged at HEAD, will not push a release commit."
           else
-            echo "::set-output name=can-tag::true"
+            echo "can_bump_version=true" >> $GITHUB_OUTPUT
```
Comment (jkonecny12) [inline .github/workflows/release-periodically.yml:50]: Interesting, I wasn't aware of this solution.

**Review by jkonecny12 — APPROVED** [review]
Looks good to me.

File: .github/workflows/release-periodically.yml (hunk 1 above)
Comment (poncovka) [reply .github/workflows/release-periodically.yml:50]: The solution with `set-output` is deprecated: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

Comment (jkonecny12) [reply .github/workflows/release-periodically.yml:50]: Good to know. Thanks!

**Pull request merged**